src/main/java/net/sf/jabref/gui/entryeditor/EntryEditorTab.java

```diff
@@ -285,9 +301,37 @@ public void setEntry(BibEntry entry) {
         try {
             updating = true;
             for (FieldEditor editor : editors.values()) {
-                String toSet = entry.getField(editor.getFieldName()).orElse("");
-                if (!toSet.equals(editor.getText())) {
-                    editor.setText(toSet);
+                String fieldName = editor.getFieldName();
+
+                boolean displayableFieldValueExists;
+                boolean hiddenFieldValueExists;
+                boolean currentFieldNameIsHidden = fieldName.startsWith("_");
+                if (currentFieldNameIsHidden) {
+                    displayableFieldValueExists = !entry.getField(fieldName.substring(1)).orElse("").isEmpty();
```

Comment:
If I think of it you could also extract ``!entry.getField(fieldName.substring(1)).orElse("").isEmpty()`` to a new method ``entryHasFieldName(entry, fieldName)``